src/common/cockpitwebserver.c

```diff
@@ -1085,8 +1093,13 @@ on_accept_certificate (GTlsConnection *conn,
                        GTlsCertificateFlags errors,
                        gpointer user_data)
 {
-  /* Only used during testing */
-  g_assert (cockpit_webserver_want_certificate == TRUE);
+  /* Only accept unknown CAs, e. g. for passing on the certificate to sssd */
+  if (errors != 0 && errors != G_TLS_CERTIFICATE_UNKNOWN_CA) {
+     g_warning ("Rejecting bad client certificate");
```

Comment:
g_warning() are for places where the Cockpit itself is in a broken state. ... lets use g_message() here.